dvc/_debug.py

```diff
@@ -5,6 +5,34 @@
     from argparse import Namespace
 
 
+@contextmanager
+def yappi_profile(save_output: bool = False, wall_clock: bool = True):
+    try:
+        import yappi  # pylint: disable=import-error
+    except ImportError:
+        print("Failed to run profiler, yappi is not installed")
+        yield
+        return
+
+    import time
+
+    yappi.set_clock_type("wall" if wall_clock else "cpu")
+
+    yappi.start()
+    yield
+    yappi.stop()
+
+    if save_output:
+        stats = yappi.get_func_stats()
+        uid = time.strftime("%Y%m%d_%H%M%S")
+        stats.save(f"callgrind.dvc-{uid}.out", "callgrind")
```

Comment:
Went autogenerated way with this yappi, let me know if you want to use `--cprofile-dump dump.prof` like flags 😄.

Reply:
Might need that option for https://github.com/iterative/dvc-bench/issues/301 , so as long as we could introduce it later - it is fine.

Reply:
Let me work on that then. :)